pkg/storage/s3.go

```diff
@@ -307,13 +363,20 @@ func (rs *S3Storage) FileExists(ctx context.Context, file string) (bool, error)
 // The first argument is the file path that can be used in `Open`
 // function; the second argument is the size in byte of the file determined
 // by path.
-func (rs *S3Storage) WalkDir(ctx context.Context, fn func(string, int64) error) error {
+func (rs *S3Storage) WalkDir(ctx context.Context, dir string, listCount int64, fn func(string, int64) error) error {
 	var marker *string
+	prefix := rs.options.Prefix
+	if len(dir) > 0 {
+		prefix += dir
+	}
```

Comment:
```suggestion
	prefix := rs.options.Prefix + dir
```

(or should we use `path.Join`? 🤔)

Reply:
I think `path.Join` is not proper. since we hardcode [prefix](https://github.com/pingcap/br/blob/master/pkg/storage/s3.go#L214)  before